unittests/test_containers.py

```diff
@@ -58,49 +67,81 @@ def create_container_platform(self):
         return containers.Docker()
 
     @property
-    def exp_cmd_mount_points(self):
+    def expected_cmd_mount_points(self):
         return ('docker run --rm -v "/path/one":"/one" -v "/path/two":"/two" '
                 "name:tag bash -c 'cd /stagedir; cmd1; cmd2'")
 
     @property
-    def exp_cmd_custom_registry(self):
+    def expected_cmd_prepare(self):
+        return []
+
+    @property
+    def expected_cmd_with_run_opts(self):
         return ('docker run --rm -v "/path/one":"/one" '
-                'registry/custom/name:tag '
-                "bash -c 'cd /stagedir; cmd'")
+                "name:tag bash -c 'cd /stagedir; cmd'")
 
 
 class TestShifterNG(_ContainerPlatformTest, unittest.TestCase):
     def create_container_platform(self):
         return containers.ShifterNG()
 
     @property
-    def exp_cmd_mount_points(self):
+    def expected_cmd_mount_points(self):
+        return ('shifter run '
+                '--mount=type=bind,source="/path/one",destination="/one" '
+                '--mount=type=bind,source="/path/two",destination="/two" '
+                "name:tag bash -c 'cd /stagedir; cmd1; cmd2'")
+
+    @property
+    def expected_cmd_prepare(self):
+        return ['shifter pull name:tag']
+
+    @property
+    def expected_cmd_with_run_opts(self):
+        self.container_platform.requires_mpi = True
```

Comment:
You still don't test the case without MPI. My suggestion was the following:

```python
class TestShifterNG(_ContainerPlatformTest, unittest.TestCase):
    # as of now, but without setting the MPI flag.

class TestShifterNGWithMPI(TestShifterNG):
    @property
    def expected_cmd_with_run_opts(self):
        self.container_platform.requires_mpi = True
        return ...
```

You should do the same for the CUDA option of Singularity.